File: backend/models/inquiryModel.js
const mongoose = require('mongoose');

const InquirySchema = new mongoose.Schema({
  gymId: { type: mongoose.Schema.Types.ObjectId, ref: 'gym', required: true },
  name: { type: String, required: true },
  phone: { type: String, required: true },
  email: { type: String },
  gender: { type: String, enum: ['Male', 'Female', 'Other'] },
  inquiryDate: { type: Date, default: Date.now },
  planToJoinDate: { type: Date },
  address: { type: String },
  status: { type: String, enum: ['Pending', 'Joined', 'Cancelled'], default: 'Pending' },
  createdAt: { type: Date, default: Date.now },
});

module.exports = mongoose.model('inquiry', InquirySchema);

File: frontend/lib/data/models/inquiry_model.dart
class InquiryModel {
  final String id;
  final String name;
  final String phone;
  final String? email;
  final String? gender;
  final DateTime inquiryDate;
  final DateTime? planToJoinDate;
  final String? address;
  final String status;

  InquiryModel({
    required this.id,
    required this.name,
    required this.phone,
    this.email,
    this.gender,
    required this.inquiryDate,
    this.planToJoinDate,
    this.address,
    required this.status,
  });

  factory InquiryModel.fromJson(Map<String, dynamic> json) {
    return InquiryModel(
      id: json['_id'] ?? '',
      name: json['name'] ?? '',
      phone: json['phone'] ?? '',
      email: json['email'],
      gender: json['gender'],
      inquiryDate: json['inquiryDate'] != null 
          ? DateTime.parse(json['inquiryDate']) 
          : DateTime.now(),
      planToJoinDate: json['planToJoinDate'] != null 
          ? DateTime.parse(json['planToJoinDate']) 
          : null,
      address: json['address'],
      status: json['status'] ?? 'Pending',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'phone': phone,
      'email': email,
      'gender': gender,
      'inquiryDate': inquiryDate.toIso8601String(),
      'planToJoinDate': planToJoinDate?.toIso8601String(),
      'address': address,
      'status': status,
    };
  }
}
